pkg/blockbuilder/blockbuilder.go

```diff
@@ -40,9 +41,13 @@ type BlockBuilder struct {
 	register    prometheus.Registerer
 	limits      *validation.Overrides
 	kafkaClient *kgo.Client
-	parts       []int32
 	bucket      objstore.Bucket
 
+	parts []int32
+	// fallbackOffset is the low watermark from where a partition which doesn't have a commit will be consumed.
+	// TODO(v): to support for setting it to kafkaStartOffset we need to rework how a lagging cycle is chopped into time frames.
+	fallbackOffset int64
```

Comment:
Maybe appropriate to name it like this because we are doing stuff like `time.UnixMilli(b.fallbackOffset)`
```suggestion
	fallbackTimeOffset int64
```

Reply:
I'm not sure about this one: the comment for the `fallbackOffset` mentions, that it is expected to be either a special `kafkaStartOffset` or a timestamp (in the future). Thus, the code should expect it to be a non-time value. I've expanded the comment to be more explicit about that.